core/src/program/blocks/span_block.rs

```diff
@@ -340,6 +334,24 @@ fn batch_ops(ops: Vec<Operation>) -> (Vec<OpBatch>, Digest) {
     (batches, hash)
 }
 
+/// Returns a CodeBlock [Span] from sequence of Span blocks provided as input.
+pub fn combine_spans(span_vec: &mut Vec<Span>) -> CodeBlock {
+    // This is to avoid redoing the same exact compution already done earlier when
+    // creating a SPAN block of operational batches from exported procedure body.
+    if span_vec.len() == 1 {
+        return CodeBlock::new_codeblock_from_span(span_vec.remove(0));
+    }
+
+    let mut ops = Vec::<Operation>::new();
+    for span in (*span_vec).iter() {
+        for batch in &span.op_batches {
+            ops.extend_from_slice(&batch.ops);
+        }
+    }
+    span_vec.clear();
+    CodeBlock::Span(Span::new(ops))
+}
+
```

Comment:
Let's do the following here:
1. Move it to `parsers/blocks.rs` file under `combine_blocks()` function.
2. Change the signature to `fn combine_spans(spans: &mut Vec<CodeBlock>) -> CodeBlock`
3. Handle the unwrapping of blocks into spans here (and panic if a non-span block is encountered).

Reply:
Done